tests/test_search_rollup.py

```diff
@@ -0,0 +1,285 @@
+import functools
```

Comment:
Overall these tests look good. Nit: they don't address the relationship between NULL and aggregation, which has been the dominant source of bugs.